### PR TITLE
[Console] fix section output when multiples section with max height

### PR DESCRIPTION
**Pull request by joelwurtz**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Partial fix for https://github.com/symfony/symfony/issues/35012
| License       | MIT
| Doc PR        | 

This pr does not fix https://github.com/symfony/symfony/issues/35012

However it fixes the second script wrote by @lyrixx (https://github.com/symfony/symfony/issues/35012#issuecomment-1573145508) with fiber and parallel output.

In short when there was multiple sections with max height, it was not taking into account the max height of other sections when doing a full rewrite, leading to erase more lines than intended.

### CODE REVIEW

File: src/Symfony/Component/Console/Tests/Output/ConsoleSectionOutputTest.php [hunk 1]
```diff
@@ -133,6 +133,40 @@ public function testMaxHeight()
...
+        $expected .= 'Two'.\PHP_EOL.'Three'.\PHP_EOL.'Four'.\PHP_EOL;
+
+        // cause overflow of first section (redraw whole section, without first line)
+        $firstSection->writeln("Four\nFive\nSix");
+        $expected .= "\x1b[6A\x1b[0J";
+        $expected .= 'Four'.\PHP_EOL.'Five'.\PHP_EOL.'Six'.\PHP_EOL;
+        $expected .= 'Two'.\PHP_EOL.'Three'.\PHP_EOL.'Four'.\PHP_EOL;
+
+        rewind($this->stream);
+        $this->assertEquals(escapeshellcmd($expected), escapeshellcmd(stream_get_contents($this->stream)));
```
Comment (joelwurtz) [inline src/Symfony/Component/Console/Tests/Output/ConsoleSectionOutputTest.php:167]: escallesshellcmd is intended here, it's way better to debug output with this, as we can properly see escaped character when there is a failure : 

image

**Review by lyrixx — APPROVED** [review]
Works great! Thanks!

[console-section.webm](https://github.com/symfony/symfony/assets/408368/82e5d012-058d-4261-b907-9e9677096380)

**Review by stof — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @joelwurtz.
